src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -811,3 +814,114 @@ fn index_of(input_string: String, search_char: char) -> i32
     }
 }
 
+#[derive(Clone)]
+struct Detection {
+    start: u64,
+    end: u64,
+    length: usize,
+    redaction_token: String
+}
+
+pub struct SecretMasker {
+    pub scan: identifiable_scans::Scan
+}
+
+impl SecretMasker {
+    pub fn mask_secrets(&mut self, input: &str, default_redaction_token: Option<&str>, validate_checksum: bool) -> String {
+        if input.is_empty() {
+            return String::new();
+        }
+
+        self.scan.reset();
+
+        let input_as_bytes = input.as_bytes();
+        self.scan.parse_bytes(input_as_bytes);
+
+        let detections = self.scan.possible_matches();
+        let mut detections = detections.clone();
```

Comment:
Probably don't need to clone if it's empty, which should increase the speed very slightly. Just calling it out.

Reply:
Valid point. Fixed.